server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2429,8 +2429,8 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
     if (!startMetadataMatchesExisting) {
       // Not in the desired start state.
       return new DataStoreMetadataUpdateResult(true, false, StringUtils.format(
-          "Inconsistent metadata state. This can happen if you update input topic in a spec without changing " +
-          "the supervisor name. Stored state: [%s], Target state: [%s].",
+          "Inconsistent metadata state. This can happen when the input topic in the supervisor spec is updated"
+          + " without resetting the supervisor. Stored state: [%s], Target state: [%s].",
```

Comment:
Instead of saying, "this can happen when ...", should we clearly call out the possible remedy like "try doing ..." ?
```suggestion
          "Inconsistency between stored metadata state[%s] and target state[%s]. Try resetting the supervisor.",
```
Also, `StringUtils.format()` is not needed here as the `DataStoreMetadataUpdateResult` takes a msg format and args.